File: back/src/main/java/com/LojaVirtual/services/CidadeService.java
package com.LojaVirtual.services;

import java.util.List;
import java.util.NoSuchElementException;
import java.util.Date;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import com.LojaVirtual.entities.Cidade;
import com.LojaVirtual.repositories.CidadeRepository;

@Service
public class CidadeService {

    @Autowired
    private CidadeRepository cidadeRepository;

    public Page<Cidade> buscarTodos(Pageable pageable) {
        return cidadeRepository.findAll(pageable);
    }

    public Cidade buscarPorId(Long id) {
        return cidadeRepository.findById(id).get();
    }

    public Cidade inserir(Cidade cidade) {
        cidade.setDataCriacao(new Date());
        Cidade cidadeNovo = cidadeRepository.saveAndFlush(cidade);
        return cidadeNovo;
    }

    public Cidade alterar(Cidade cidade) {
        cidade.setDataAtualizacao(new Date());
        return cidadeRepository.saveAndFlush(cidade);
    }

    public void excluir(Long id) {
        Cidade cidade = cidadeRepository.findById(id)
                .orElseThrow(() -> new NoSuchElementException("Cidade não encontrada."));
        cidadeRepository.delete(cidade);
    }
}

File: frontendloja/src/services/CidadeService.js
import axios from 'axios';

export class CidadeService{
	URL = "http://localhost:8080/api/cidade";

	inserir(cidade){
		return axios.post(this.URL, cidade);
	}

	alterar(cidade){
		return axios.put(this.URL, cidade);
	}

	excluir(id){
		return axios.delete(this.URL+"/"+id);
	}

	listar(page,size){
		return axios.get(this.URL + "?page=" + page + "&size=" + size);
	}
}